utils/decorators.py

```diff
@@ -0,0 +1,25 @@
+from decorator import decorator
+
+@decorator
+def members_only(func, *args, **kwargs):
+
+    def blocker(*args, **kwargs):
+        return 'You need to be a member of this ' \
+               'organization to use this command.'
+
+    # plugin instance
+    instance = args[0]
+    msg = args[1]
+    user = msg.frm.nick
+
+    teams = [instance.TEAMS[instance.GH_ORG_NAME + ' newcomers'],
+             instance.TEAMS[instance.GH_ORG_NAME + ' developers'],
+             instance.TEAMS[instance.GH_ORG_NAME + ' maintainers']
+            ]
+
+    for team in teams:
+        if team.is_member(user):
+            return func(*args, **kwargs)
+
+    return blocker(*args, **kwargs)
```

Comment:
Why not use something simpler and functionally equivalent, like : 
```py
def members_only(func):
    def wrapper(*args, **kwargs):
        instance = args[0]
        msg = args[1]
        user = msg.frm.nick
        teams = [
            instance.TEAMS[instance.GH_ORG_NAME + ' newcomers'],
            instance.TEAMS[instance.GH_ORG_NAME + ' developers'],
            instance.TEAMS[instance.GH_ORG_NAME + ' maintainers'],
        ]

        for team in teams:
            if team.is_member(user):
                return func(*args, **kwargs)
        
        return 'You need to be a member of this organisation to use this command.'
```

Reply:
I did tried this way and many other ways similar so that I don't have to use `decorator` but Errbot was not accepting that as a bot command and was resulting in 'Command not found' error.